### PR TITLE
Use the AzApi provider rather than wrapping the ARM template

### PR DESCRIPTION
**Pull request by ilmax**
I think the AzApi provider is a the better option to use here, it's less verbose due to not requiring the ARM file. This should not have any meaningful change and the outcome should be the same. I use this exact resource to create the GitHub network setting resource in my environment and it works just fine.

Also I fixed the variable names in the example so terraform validate now works on both of them.

P.S. Thanks for putting this together :)

### CODE REVIEW

**Comment by pwideman** [comment]
Thanks for the contributions @ilmax. Couple of things here:

1. There are actually a few more places where var names need to be fixed up to align to the new "business" ID instead of enterprise ID. I also think we should keep those changes separate from your more important AzApi changes, so we'll have a cleaner commit history and such. I will make those changes in a separate PR.
2. This repo contains two TF modules, in `/modules/nsg` and `/modules/firewall`. Your AzApi changes need to be made in both modules, not just the firewall module.

Otherwise, I agree that using the AzApi resource provider is cleaner than the ARM template. Have you verified that `terraform destroy` will delete the `networkSettings` resource? I will try this out myself as well.

File: examples/example-firewall/main.tf [hunk 1]
```diff
@@ -3,5 +3,5 @@ module "vnet" {
     source = "github.com/garnertb/github-runner-vnet//modules/firewall"
 
     base_name = var.base_name
-    github_enterprise_id = var.github_enterprise_id
+    github_business_id = var.github_enterprise_id
```
Comment (pwideman) [inline examples/example-firewall/main.tf:6]: There's a few other places where similar changes need to be made, I've created another PR to handle these so let's roll these back here.

File: examples/example-nsg/main.tf [hunk 2]
```diff
@@ -3,5 +3,5 @@ module "vnet" {
     source = "github.com/garnertb/github-runner-vnet//modules/nsg"
 
     base_name = var.base_name
-    github_enterprise_id = var.github_enterprise_id
+    github_business_id = var.github_enterprise_id
```
Comment (pwideman) [inline examples/example-nsg/main.tf:6]: See above - let's roll this back here and I'll submit a more comprehensive fix.

File: modules/firewall/main.tf [hunk 3]
```diff
@@ -235,20 +240,21 @@ resource "azurerm_subnet_route_table_association" "runner_subnet_route_table_ass
...
+      businessId = var.github_business_id
+      subnetId   = azurerm_subnet.runner_subnet.id
+    }
   })
-  template_content    = file("${path.module}/../shared/gh_network_settings_template.json")
+  response_export_values = ["tags.GitHubId"]
+
+  lifecycle {
+    ignore_changes = [tags]
+  }
```
Comment (pwideman) [inline modules/firewall/main.tf:259]: These changes look good but the `/modules/nsg` module also needs to be updated with the same changes.

**Review by pwideman — CHANGES_REQUESTED** [review]

**Comment by ilmax** [comment]
> This repo contains two TF modules, in /modules/nsg and /modules/firewall. Your AzApi changes need to be made in both modules, not just the firewall module.

I somehow missed this 🤷 it's fixed now. Let me know if you prefer me to rebase the changes into a single commit.

> This repo contains two TF modules, in /modules/nsg and /modules/firewall. Your AzApi changes need to be made in both modules, not just the firewall module.

Honestly nope, but in the worst case, we can use something akin to [this](https://github.com/Azure/terraform-provider-azapi/issues/314) to force a deletion when issuing `terraform destroy`

**Comment by pwideman** [comment]
@ilmax no need to rebase, we will squash and merge anyway.

The code changes look good, but deletion of the `GitHub.Network/networkSettings` resources needs to be confirmed before we can accept this. I still plan on testing it myself but not sure when I will find time.

File: modules/firewall/outputs.tf [hunk 4]
```diff
@@ -1,4 +1,4 @@
 output "network_settings_id" {
     description = "ID of the GitHub.Network/networkSettings resource"
-    value = jsondecode(azurerm_resource_group_template_deployment.github_network_settings.output_content).gitHubId.value
+    value = jsondecode(azapi_resource.github_network_settings.output).gitHubId.value
```
Comment (pwideman) [inline modules/firewall/outputs.tf:3]: This is wrong and results in a runtime error, it should be:
```suggestion
    value = jsondecode(azapi_resource.github_network_settings.output).tags.GitHubId
```

File: modules/nsg/outputs.tf [hunk 5]
```diff
@@ -1,4 +1,4 @@
 output "network_settings_id" {
     description = "ID of the GitHub.Network/networkSettings resource"
-    value = jsondecode(azurerm_resource_group_template_deployment.github_network_settings.output_content).gitHubId.value
+    value = jsondecode(azapi_resource.github_network_settings.output).gitHubId.value
```
Comment (pwideman) [inline modules/nsg/outputs.tf:3]: This is wrong and results in a runtime error, it should be:
```suggestion
    value = jsondecode(azapi_resource.github_network_settings.output).tags.GitHubId
```

File: modules/nsg/main.tf
Comment (pwideman) [inline modules/nsg/main.tf:1]: This file needs the `azapi` required provider just like `modules/firewall/main.tf`:

```
    azapi = {
      source  = "Azure/azapi"
      version = "~> 1.14.0"
    }
```

**Review by pwideman — CHANGES_REQUESTED** [review]
I tested this, creation and delete. Delete works fine with how `azapi` is currently being used, nothing else to do there. However there were validation and runtime errors that should have been readily apparent with any testing. I've provided comments to show what needs to be changed to fix those errors.

Also, your `main` branch is not up to date with `garnertb:main` after I fixed the business/enterprise ID variable issues. It shouldn't cause a problem for this PR because the differences do not overlap, but you still probably want to update your repo.

**Comment by ilmax** [comment]
Thanks for checking this one out, let me resolve the issues that are a copy-paste error.

**Comment by pwideman** [comment]
> I apologize for the issue that was a copy-paste error, I didn't apply this configuration, I just backported some of the changes I made because I thought it could improve this repo 🤷

No worries there, they are good contributions that are much appreciated, and we'll get the kinks worked out.

**Review by pwideman — APPROVED** [review]

**Pull request merged**